scripts/research/b2_02_boundary_interaction_path_lib.py

```
@@ -0,0 +1,978 @@
+                "candidate_count": len(candidate_queue),
+            }
+        )
+        scored.append(record)
+
+        path_count = sum(
+            1
+            for item in base_queue
+            if int(item[2]) == int(event["path_state"])
+        )
+        if path_count < CANDIDATE_MIN_COUNT:
+            raise B202Error("placebo stratum count diverged from candidate count")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)